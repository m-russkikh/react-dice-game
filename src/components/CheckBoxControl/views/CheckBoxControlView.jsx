import React from 'react';
import { FormGroup, Label, Input } from 'reactstrap';

import { CheckBoxControlViewPropType } from '../propTypes';


function CheckBoxControlView({ label, onChange, disabled }) {
  return (
    <FormGroup check>
      <Label check>
        <Input
          type="checkbox"
          onChange={onChange}
          disabled={disabled}
        />
        { label }
      </Label>
    </FormGroup>
  );
}

CheckBoxControlView.propTypes = CheckBoxControlViewPropType;

CheckBoxControlView.defaultProps = {
  onChange: () => {},
  disabled: false,
};

export default CheckBoxControlView;
