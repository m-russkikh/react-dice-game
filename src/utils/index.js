import { MAX_BET_NUMBER } from '../constants/diceGame';


export function saveState(state) {
  const serializedBalance = JSON.stringify(state.balance);

  localStorage.setItem('balance', serializedBalance);
}

export function getBetNumber() {
  return Math.floor(Math.random(0, 1) * MAX_BET_NUMBER) + 1;
}
