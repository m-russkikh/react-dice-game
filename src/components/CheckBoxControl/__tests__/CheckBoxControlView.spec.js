import { getShallowWrapper } from '../../../utils/testUtils';
import CheckBoxControlView from '../views/CheckBoxControlView';


describe('CheckBoxControlView', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      label: 'label',
      disabled: false,
      onChange: jest.fn(),
    };

    wrapper = getShallowWrapper(CheckBoxControlView, props);
  });

  describe('render', () => {
    it('should display correct', () => {
      expect(wrapper.debug()).toMatchSnapshot();
    });
  });

  describe('Input', () => {
    describe('change event', () => {
      it('should handle change event', () => {
        const event = { currentTarget: { checked: true } };

        wrapper.find('Input').simulate('change', event);

        expect(props.onChange).toHaveBeenCalledTimes(1);
        expect(props.onChange).toHaveBeenCalledWith(event);
      });
    });

    describe('disabled', () => {
      it('should be disabled', () => {
        wrapper.setProps({ ...props, disabled: true });

        expect(wrapper.find('Input').prop('disabled')).toBeTruthy();
      });

      it('should be enabled', () => {
        wrapper.setProps({ ...props, disabled: false });

        expect(wrapper.find('Input').prop('disabled')).toBeFalsy();
      });
    });
  });
});
