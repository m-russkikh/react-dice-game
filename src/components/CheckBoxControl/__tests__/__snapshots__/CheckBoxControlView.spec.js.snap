// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxControlView render should display correct 1`] = `
"<FormGroup check={true} tag=\\"div\\">
  <Label check={true} tag=\\"label\\" widths={{...}}>
    <Input type=\\"checkbox\\" onChange={[Function: mockConstructor]} disabled={false} />
    label
  </Label>
</FormGroup>"
`;
